opensearch-operator/pkg/builders/cluster.go

```diff
@@ -188,6 +188,17 @@ func NewSTSForNodePool(
 	readinessProbeTimeoutSeconds := int32(30)
 	readinessProbeFailureThreshold := int32(5)
 	readinessProbeInitialDelaySeconds := int32(60)
+	readinessProbeCommand := []string{
+		"/bin/bash",
+		"-c",
+		// In the readiness check wait for all shards to be initialized before reporting ready.
+		// Otherwise PDB would allow another pod to be terminated while cluster potentially is still in yellow state due to initializing shards.
+		// Use local=true to check only the local node
+		// Use wait_for_no_initializing_shards=true to ensure that all shards are initialized
+		fmt.Sprintf("curl -k -u \"$(cat /mnt/admin-credentials/username):$(cat /mnt/admin-credentials/password)\" --silent --fail 'https://localhost:%d/_cluster/health?local=true&wait_for_no_initializing_shards=true'", PortForCluster(cr)),
+		// TODO: Change default parameters or keep original state?
+		// fmt.Sprintf("curl -k -u \"$(cat /mnt/admin-credentials/username):$(cat /mnt/admin-credentials/password)\" --silent --fail 'https://localhost:%d'", PortForCluster(cr)),
```

Comment:
Open TODO to be discussed here.
Should the parameters `?local=true&wait_for_no_initializing_shards=true` be added per default?

From docs it seems that parameters are also available in 1.x
https://opensearch.org/docs/1.3/api-reference/cluster-api/cluster-health/

This seems like a safe change and improvement for users in general.
But maybe it also just an issue I'm having due to many eviction events in my clusters.

Both ways are fine for me. Can override it via CRD.

Reply:
I have to correct myself. The ReadinessProbe is the wrong place. It should be the startup probe.
I made the startup probe an command as well. With the same command as the readiness probe.

Additionally the `local=true` in combination with `wait_for_no_initializing_shards=true` doesn't seem to behave the way I thought. Still all nodes would report `Unhealthy` readiness probes.

I'm also unable to cover all possible scenarios how people deploy and configure their clusters.
Therefore I decided to leave the default in place. Whoever needs this is then able to adjust the probes in the CR.

Will soon push the changed and tested code.